massa-bootstrap/src/server.rs

```diff
@@ -205,35 +183,34 @@ pub fn start_bootstrap_server<D: Duplex + 'static>(
     }))
 }
 
-struct BootstrapServer<'a, D: Duplex> {
+struct BootstrapServer<'a, C: NetworkCommandSenderTrait> {
```

Comment:
You are generic over `NetworkCommandSenderTrait` to allow to be used with mocks ? In my mind, that seems a bit too much just for a mock because imagine you had interactions with 10 modules you would have to pass 10 generic type but in fact behind nothing is generic it's just mock.

I don't have an idea on the best practice but I want to raise this point

Reply:
I haven't given it too much thought, and agree that it can open up a can of worms.

In function definitions, it might be better be generic in the argument (impl Thing + Send + ...), But at the struct level, it means using Box<dyn Thing + ...> Which we probably want to avoid.

Can't get to a laptop for now, but I'll see what can be done to clean things up

Reply:
I've put some thought into it, and played around, and it seems to come down to the fact that a mock-implementation is fundamentally a different type to the unmocked. This creates a need to use the thing being mocked via a trait with the current design.

I've asked a question about this in the library discussions: https://github.com/asomers/mockall/discussions/478 with any luck, we will be able to see a solution that allows us to mock structs without mangling names

Would you be happy with continuing with this PR, with the understanding that we consider your point in any future PRs?

Reply:
managed to work out how to do it cleanly: [ce04b9e](https://github.com/massalabs/massa/pull/3745/commits/ce04b9e91ce91551f25ed8542765b4249ffeb59a)

Reply:
Very nice like that thank you !